src/Runner.Listener/JobDispatcher.cs

```diff
@@ -840,90 +969,93 @@ private async Task TryUploadUnfinishedLogs(Pipelines.AgentJobRequestMessage mess
+                        var record = logRecords[pages.Key];
+                        if (record.Log == null)
+                        {
+                            // Create the log
+                            record.Log = await jobServer.CreateLogAsync(message.Plan.ScopeIdentifier, message.Plan.PlanType, message.Plan.PlanId, new TaskLog(String.Format(@"logs\{0:D}", record.Id)), default(CancellationToken));
 
-                        // Need to post timeline record updates to reflect the log creation
-                        updatedRecords.Add(record.Clone());
-                    }
+                            // Need to post timeline record updates to reflect the log creation
+                            updatedRecords.Add(record.Clone());
+                        }
 
-                    for (var i = 1; i <= pages.Value.Count; i++)
-                    {
-                        var logFile = pages.Value[i];
-                        // Upload the contents
-                        using (FileStream fs = File.Open(logFile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                        for (var i = 1; i <= pages.Value.Count; i++)
                         {
-                            var logUploaded = await jobServer.AppendLogContentAsync(message.Plan.ScopeIdentifier, message.Plan.PlanType, message.Plan.PlanId, record.Log.Id, fs, default(CancellationToken));
+                            var logFile = pages.Value[i];
+                            // Upload the contents
+                            using (FileStream fs = File.Open(logFile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                            {
+                                var logUploaded = await jobServer.AppendLogContentAsync(message.Plan.ScopeIdentifier, message.Plan.PlanType, message.Plan.PlanId, record.Log.Id, fs, default(CancellationToken));
+                            }
+
+                            Trace.Info($"Uploaded unfinished log '{logFile}' for current job.");
+                            IOUtil.DeleteFile(logFile);
                         }
+                    }
 
-                        Trace.Info($"Uploaded unfinished log '{logFile}' for current job.");
-                        IOUtil.DeleteFile(logFile);
+                    if (updatedRecords.Count > 0)
+                    {
+                        await jobServer.UpdateTimelineRecordsAsync(message.Plan.ScopeIdentifier, message.Plan.PlanType, message.Plan.PlanId, message.Timeline.Id, updatedRecords, CancellationToken.None);
                     }
                 }
-
-                if (updatedRecords.Count > 0)
+                else
                 {
-                    await jobServer.UpdateTimelineRecordsAsync(message.Plan.ScopeIdentifier, message.Plan.PlanType, message.Plan.PlanId, message.Timeline.Id, updatedRecords, CancellationToken.None);
+                    Trace.Info("Job server does not support log upload yet.");
```

Comment:
This will be handled by results team folks later, so ignoring that and acting as a placeholder for now